File: src/main/java/alfheim/api/event/TileUpdateEvent.java
package alfheim.api.event;

import cpw.mods.fml.common.eventhandler.Cancelable;
import cpw.mods.fml.common.eventhandler.Event;
import net.minecraft.tileentity.TileEntity;

@Cancelable
public class TileUpdateEvent extends Event {

	public final TileEntity tile;
	
	public TileUpdateEvent(TileEntity te) {
		tile = te;
	}
	
	/** Used in ASM */
	public static TileUpdateEvent instantiate(TileEntity te) {
		return new TileUpdateEvent(te);
	}
	
	/** Used in ASM */
	public static void stub() {}
}

File: src/main/java/alfheim/api/event/TileUpdateEvent.kt
package alfheim.api.event

import cpw.mods.fml.common.eventhandler.*
import net.minecraft.tileentity.TileEntity

@Cancelable
class TileUpdateEvent(val tile: TileEntity): Event() {
	
	companion object {
		
		/** Used in ASM  */
		@JvmStatic
		fun instantiate(te: TileEntity): TileUpdateEvent {
			return TileUpdateEvent(te)
		}
		
		/** Used in ASM  */
		@JvmStatic
		fun stub() {}
	}
}
